kedro-org/kedro-viz
# Migrate Viz to use Pydantic V2

## Description

Resolves #1603 

## Development notes

* Followed [pydantic migration guide](https://docs.pydantic.dev/latest/migration/ )
* Partially updated code using `bump-pydantic` tool
* Manually updated `@validators` to `@field_validators`
* Updated requirements.txt with new pydantic version
* Updated test files as per lint suggestions

## QA notes

* All pytests should pass and `kedro viz run` should work as expected

**Note**: Observed flowchart node arrangements change (mirrored) due to the migration, though the data remains un-affected as below -

**Before Migration:**
 
<img width="1262" alt="current_placement" src="https://github.com/kedro-org/kedro-viz/assets/87735534/3a0f1c85-6adb-436f-a726-db3fe4355372">

**After Migration:**
<img width="1239" alt="pydantic_v2_placement" src="https://github.com/kedro-org/kedro-viz/assets/87735534/77c2d2f3-b95d-4842-badc-f12b9105536b">


## Checklist
